yt_dlp/extractor/dlf.py

```diff
@@ -0,0 +1,209 @@
+import re
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    extract_attributes,
+    int_or_none,
+    url_or_none,
+)
+
+
+class DLFBaseIE(InfoExtractor):
+    _VALID_URL_BASE = r'https?://(?:www\.)?deutschlandfunk\.de/'
+    _BUTTON_REGEX = r'(<button[^>]+alt="Anhören"[^>]+data-audio-diraid[^>]*>)'
+
+    def _parse_button_attrs(self, button, id=None):
+        attrs = extract_attributes(button)
+        id = id or attrs['data-audio-diraid']
```

Comment:
Avoid redefining built-ins (`id`)
You can rename the var to whatever you want, but for the sake of this review I will be using `id_` throughout
```suggestion
    def _parse_button_attrs(self, button, id_=None):
        attrs = extract_attributes(button)
        id_ = id_ or attrs['data-audio-diraid']
```